src/util/tubemap.test.js

```diff
@@ -0,0 +1,214 @@
+            ]
+          }
+        ]
+      };
+      expect(cigar_string(readPath)).toBe("32M14I10M40D");
+    });
+
+    it('it can handle a mapping with multiple edits', async () => {
+      const readPath = {
+        mapping: [
+          {
+            edit: [
+              {
+                from_length: 8,
+                to_length: 9
+              }
+            ]
+          },
+          {
+            edit: [
+              {
+                from_length: 2,
+                to_length: 1
+              }
+            ]
+          },
+          {
+            edit: [
+              {
+                from_length: 4,
+                to_length: 4
+              }
+            ]
+          },
+          {
+            edit: [
+              {
+                from_length: 18,
+                to_length: 11
+              }
+            ]
+          },
+          {
+            edit: [
+              {
+                from_length: 15,
+                sequence: "C"
+              }
+            ]
+          },
+          {
+            edit: [
+              {
+                from_length: 3,
+                to_length: 4,
+                sequence: "G"
+              }
+            ]
+          }, 
+          {
+            edit: [
+              {
+                from_length: 8,
+                to_length: 9,
+                sequence: "C"
+              }
+            ]
+          }, 
+          {
+            edit: [
+              {
+                from_length: 10,
+                to_length: 10,
+                sequence: "ATCG"
+              }
+            ]
+          },
+          {
+            edit: [
+              {
+                from_length: 15,
+                to_length: 14,
+                sequence: "G"
+              }
+            ]
+          },
+          {
+            edit: [
+              {
+                to_length: 9,
+                sequence: "AAAA"
+              }
+            ]
+          }
+        ]
+      };
+      expect(cigar_string(readPath)).toBe("8M1I1M1D15M22D3M1I8M1I24M1D9I");
+    });
+
+    it('it can handle multiple edits with lengths of 0', async () => {
+      const readPath = {
+        mapping: [
+          {
+            edit: [
+              {
+                from_length: 0,
+                to_length: 0
+              }
+            ]
+          },
+          {
+            edit: [
+              {
+                from_length: 0,
+                to_length: 0
+              }
+            ]
+          },
+          {
+            edit: [
+              {
+                from_length: 0,
+                to_length: 0
+              }
+            ]
+          },
+          {
+            edit: [
+              {
+                from_length: 0,
+                to_length: 0
+              }
+            ]
+          },
+          {
+            edit: [
+              {
+                from_length: 0,
+                sequence: "C"
+              }
+            ]
```

Comment:
This doesn't actually make any sense, since if sequence is set its length always has to be `to_length`. But I suppose it is a good test case to see if we can break it.